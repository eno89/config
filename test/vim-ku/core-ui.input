" Test: vim-ku core-ui
runtime! plugin/ku.vim plugin/ku/**/*.vim
call vspec#hint({'scope': 'ku#_local_variables()', 'sid': 'ku#_sid_prefix()'})
function s:describe__ui_named_key_mappings()  "{{{1
  It requires the following initialization to test its effect

  let original_bufnr = bufnr('')
  let original_window_layout = winrestcmd()

  Should ku#start() == 1
  Should nr2char(getchar(0)) ==# 'A'
  Should nr2char(getchar(0)) ==# ''

  let ku_bufnr = bufnr('')
  let ku_window_layout = winrestcmd()

  close

  It should define <Plug>(ku-quit-session) and it works

  Should ku#start() == 1
  Should nr2char(getchar(0)) ==# 'A'
  Should nr2char(getchar(0)) ==# ''

  execute 'normal' "\<Plug>(ku-quit-session)"
  Should bufnr('') != ku_bufnr
  Should bufnr('') == original_bufnr
  Should winrestcmd() !=# ku_window_layout
  Should winrestcmd() ==# original_window_layout

  silent! close

  " FIXME: NIY: <Plug>(ku-choose-and-do-an-action)
  " FIXME: NIY: <Plug>(ku-do-the-default-action)

  " Reset side effect
  Should ku#start() == 1
  Should nr2char(getchar(0)) ==# 'A'
  Should nr2char(getchar(0)) ==# ''

  bwipeout!
endfunction




" __END__  "{{{1
" vim: filetype=vim foldmethod=marker
