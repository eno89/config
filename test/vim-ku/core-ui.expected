==== ui_named_key_mappings choose action
---- It requires the following initialization 
.....
---- It should work 
..................
---- It requires to reset side effect 
....

==== ui_named_key_mappings do default action
---- It requires the following initialization 
.....
---- It should work 
........
---- It requires to reset side effect 
....

==== ui_named_key_mappings quit session
---- It requires the following initialization 
...
---- It should work 
......
---- It requires to reset side effect 
...



**** Result ****


56 examples, 0 failures
